src/Views/Chat.hs

```diff
@@ -31,82 +35,223 @@ createChatDOMInParent api parentDiv = do
+    ]
+
+addNewChat :: LobbyAPI -> String -> Client ()
+addNewChat api chatName = do
+  addTabToTabHeader chatName
+  addChatToChatContainer chatName
+  addInputFieldToInputFieldsContainer api chatName
+
+-- | Adds a new named tab to the tabs header.
+-- | Clicks on this tabs changes changes to only show its respective chat window and input field
+addTabToTabHeader :: String -> Client ()
+addTabToTabHeader chatName =
+  "chat-tabs" `withElem` \chatTabsHeader -> do
+    chatTab <- newElem "li" `with`
+      [
+        attr "id"   =: ("chat-tab-" ++ chatName),
+        attr "role" =: "presentation"
+      ]
+    onEvent chatTab Click $ \_ -> setActiveChat chatName
+
+    textElem <- newTextElem chatName
+    linkWrapper <- newElem "a"
+    appendChild linkWrapper textElem
+    appendChild chatTab linkWrapper
+    appendChild chatTabsHeader chatTab
+
+-- | Adds a new chat window to the chat container.
+-- | The given string is used to set an id attribute.
+addChatToChatContainer :: String -> Client ()
+addChatToChatContainer chatName =
+  "chat-container" `withElem` \chatContainer -> do
+    chatDiv <- newElem "div" `with`
+      [
+        attr "id" =: ("chat-container-" ++ chatName),
+        attr "style" =: "overflow: auto; height: 200px; word-wrap: break-word;"
+      ]
+    appendChild chatContainer chatDiv
+
+-- | Adds a new input field to the inputs container.
+-- | The input field forwards the message to the handleChatInput function when 'enter' is pressed.
+addInputFieldToInputFieldsContainer :: LobbyAPI -> String -> Client ()
+addInputFieldToInputFieldsContainer api chatName =
+  "input-container" `withElem` \inputFieldContainer -> do
+    inputField <- newElem "input" `with`
+      [
+        attr "id" =: ("input-field-" ++ chatName),
+        attr "type" =: "text",
+        attr "cols" =: "60"
+
+      ]
+    onEvent inputField KeyPress $ \13 -> handleChatInput api chatName
+    appendChild inputFieldContainer inputField
+    return ()
+
+deleteChat :: String -> Client ()
+deleteChat chatName = do
+  liftIO $ print $ "chat-tab-"       ++ chatName
+  liftIO $ print $ "chat-container-" ++ chatName
+  liftIO $ print $ "input-field-"    ++ chatName
+  deleteDOM ("chat-tab-"       ++ chatName) "chat-tabs"
+  deleteDOM ("chat-container-" ++ chatName) "chat-container"
+  deleteDOM ("input-field-"    ++ chatName) "input-container"
+  return ()
+
+-- | Show the proper DOM elements for a named chat.
+-- | All other chat elements are hidden.
+setActiveChat :: String -> Client ()
+setActiveChat chatName = do
+  "chat-tabs" `withElem` \chatTabs -> do
+    setClassOnChildren chatTabs ""
+    maybeChatTab <- elemById $ "chat-tab-" ++ chatName
+    setClassOnMaybeDOMElem maybeChatTab "active"
+  "chat-container" `withElem` \chatContainer -> do
+    setClassOnChildren chatContainer "hide"
+    maybeChatContainer <- elemById $ "chat-container-" ++ chatName
+    setClassOnMaybeDOMElem maybeChatContainer ""
+  "input-container" `withElem` \inputs -> do
```

Comment:
Unless the user changes the id's in the browsers developer mode then I think this should be safe.

They are however magic constants, I'll extract them...